### PR TITLE
Browse files

### PR DESCRIPTION
**Pull request by this-pama**
- [x] Allow users to browse files.
- [x] Allow users to upload consent files.
- [x] Implement a script named `extractFileInfoFromAzureStorage`. This script is designed to interact with Azure storage to retrieve information about previously uploaded files. It works by extracting the list of blobs (files) from a specified Azure storage container. For each blob, it extracts the user UUID from the file name and checks if there is a valid user with the UUID in the database. The blobs are then categorized into two groups: known users (valid UUID) and unknown users (invalid UUID). The script then prompts the user with options to insert the records of known users, unknown users, both, or abort the operation. Based on the user's response, the appropriate records are inserted into the database. This allows us to maintain a local reference of the files already uploaded to Azure storage in our database, enhancing the efficiency of file browsing operations.

### CODE REVIEW

**Comment by myjyby** [comment]
Ok I fixed the issues/ added the points I commented on. I also finalized the consent module integration.
More generally, the consent module follows a logic of "request an external resource" (although technically this should also work for internal resources).

The idea is:
- set the url to extract resources from in a metafield (```metafields``` array) of type ```attachment``` in the config file for the platform. See example below:
```
{
  type: 'attachment',
  name: 'consent',
  required: true,
  uris: [
    {
      uri: 'https://consent.sdg-innovation-commons.org/en/contribute/resource',
    },
    { uri: undefined },
  ],
  limit: 1,
}, 
```
- this will be used to navigate to the ```uris.uri```. Note the ```uri: undefined``` enables users to input a direct link to a resource (without having to go fetch it on another platform).
- the logic is then:
- on the origin platform (e.g. the solutions mapping platform):
- request/resource 
- on the queried platform (e.g. the consent platform):
- contribute/resource (renders a page to select a resource)
- share/resource (processes the selected file and redirects to requesting origin)
- back to the origin platform
- save/resource

**Review by myjyby — DISMISSED** [review]
lgtm

File: views/contribute/pad/index.ejs [hunk 1]
```diff
@@ -204,6 +211,14 @@
 											// SO THAT AUTHORS CAN GO CHECK THEM OUT
 										}
 									</script>
+								<!-- IF THERE IS A TOKEN WITH A CALLBACK URL, THEN THIS SHOULD AUTO SUBMIT AND SEND BACK THE PAD ID TO THE REQUESTING SOURCE -->
+								<% } else if (token && action === 'publish_and_share') { %>
+									<% console.log(token) %>
```
Comment (JosuaKrause) [inline views/contribute/pad/index.ejs:216]:
```suggestion
```

File: views/contribute/pad/index.ejs [hunk 2]
```diff
@@ -204,6 +211,14 @@
...
+									<% console.log(action) %>
```
Comment (JosuaKrause) [inline views/contribute/pad/index.ejs:217]:
```suggestion
```

**Review by JosuaKrause — APPROVED** [review]
lgtm

**Pull request merged**